test/metabase/analytics/prometheus_test.clj

```diff
@@ -152,3 +154,27 @@
                                    (metric-lines port))]
         (is (seq (set/intersection expected-lines actual-lines))
             "Registry does not have c3p0 metrics in it")))))
+
+(deftest email-collector-test
+  (testing "Registry has email metrics registered"
+    (with-prometheus-system [_ system]
+      (let [sample-names (set (for [metric-family-samples (enumeration-seq (.metricFamilySamples ^CollectorRegistry (.raw ^IapetosRegistry (.-registry system))))
+                                    sample-name           (.getNames ^Collector$MetricFamilySamples metric-family-samples)]
+                                sample-name))]
+        (is (contains? sample-names "metabase_email_messages") "messages sample not found")
+        (is (contains? sample-names "metabase_email_message_errors") "message_errors sample not found")))))
+
+(defn- get-sample-value
+  [^PrometheusSystem system ^String metric]
+  (.getSampleValue ^CollectorRegistry (registry/raw (.-registry system)) metric))
+
+(deftest inc-server-test
+  (testing "inc has no effect if system is not setup"
```

Comment:
You should run these where you know the system is not set up. Want to make sure these tests run with a nil system.